### PR TITLE
Sanitize tool results and inputs in database

### PR DESCRIPTION
**Pull request by frankbria**
Implements comprehensive secret detection and scrubbing before database storage to prevent accidental exposure of sensitive data.

Features:
- Pattern-based detection for API keys (OpenAI, Anthropic, GitHub, AWS, Google, Stripe, Slack, SendGrid, npm, etc.)
- Database connection string detection (PostgreSQL, MongoDB, MySQL, Redis)
- Private key detection (RSA, EC, OpenSSH, PGP)
- Environment variable pattern detection (PASSWORD=, SECRET=, TOKEN=, etc.)
- JWT token and Bearer/Basic auth header detection
- Recursive object scrubbing for nested structures
- Audit logging of detected secret types (without exposing values)
- Optional AES-256-GCM encryption at rest

Integration:
- Scrubbing happens in hooks.ts before database storage
- Both tool-complete and notification hooks are protected
- Enabled by default (SCRUB_SECRETS=true)

Testing:
- 64 unit tests for pattern detection, edge cases, and configuration
- 17 integration tests for end-to-end hook processing
- Added uuid mock to fix Jest ESM compatibility issues

Documentation:
- Updated ARCHITECTURE.md with secret scrubbing section
- Added configuration options to .env.example

Addresses issue #10 

## Summary by CodeRabbit

* **New Features**
  * Automatic secret scrubbing to redact detected sensitive values before storage, with audit logging and optional encryption-at-rest for high-security environments; new config toggles to enable/disable these behaviors.

* **Documentation**
  * Architecture docs expanded with secret-scrubbing behavior, audit logging details, examples, and supported secret patterns.

* **Tests**
  * Extensive integration and unit tests added covering detection, redaction, audit logging, encryption, idempotency, and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/api/hooks.ts [hunk 1]
```diff
@@ -129,6 +130,42 @@ export function createHookRouter(db: Pool) {
...
+
+        // Log scrubbed secrets for audit trail
+        if (allFoundSecrets.length > 0) {
+          logScrubbedSecrets([...new Set(allFoundSecrets)], {
+            sessionId: session,
+            tool,
+            eventId,
+          });
+        }
+      }
```
[the remark this answers is not in the document]
Comment (frankbria) [reply src/api/hooks.ts]: Prompted

File: src/services/secretScrubber.ts [hunk 2]
```diff
@@ -0,0 +1,790 @@
...
+ * Result of scrubbing operation
+ */
+export interface ScrubResult {
+  /** The scrubbed text with secrets redacted */
+  scrubbed: string;
+  /** List of detected secret types (for audit logging) */
+  foundSecrets: string[];
+  /** Count of total secrets found */
+  secretCount: number;
+}
```
[the remark this answers is not in the document]
Comment (frankbria) [reply src/services/secretScrubber.ts]: Prompted

File: src/api/hooks.ts [hunk 3]
```diff
@@ -328,23 +382,30 @@ export function createHookRouter(db: Pool) {
...
         eventType,
         sessionId: resolvedSession.id,
         claudeSessionId: resolvedSession.claudeSessionId,
-        messageLength: message?.length,
+        messageLength: scrubbedMessage?.length,
         timestamp: eventTimestamp,
-        clientTimestamp: !!parsedTimestamp
+        clientTimestamp: !!parsedTimestamp,
+        secretsScrubbed: scrubbedMessage !== message
       });
```
[the remark this answers is not in the document]
Comment (frankbria) [reply src/api/hooks.ts]: Prompted
